fontbe/src/orchestration.rs

```diff
@@ -117,54 +122,85 @@ impl From<WorkId> for AnyWorkId {
     }
 }
 
+impl From<&FeWorkIdentifier> for AnyWorkId {
+    fn from(id: &FeWorkIdentifier) -> Self {
+        AnyWorkId::Fe(id.clone())
+    }
+}
+
+impl From<&WorkId> for AnyWorkId {
+    fn from(id: &WorkId) -> Self {
+        AnyWorkId::Be(id.clone())
+    }
+}
+
 /// <https://learn.microsoft.com/en-us/typography/opentype/spec/glyf>
 #[derive(Debug, Clone)]
 pub enum Glyph {
-    Simple(SimpleGlyph),
-    Composite(CompositeGlyph),
+    Simple(GlyphName, SimpleGlyph),
+    Composite(GlyphName, CompositeGlyph),
 }
 
-impl From<SimpleGlyph> for Glyph {
-    fn from(value: SimpleGlyph) -> Self {
-        Glyph::Simple(value)
+impl Glyph {
+    pub(crate) fn new_simple(glyph_name: GlyphName, simple: SimpleGlyph) -> Glyph {
+        Glyph::Simple(glyph_name, simple)
     }
-}
 
-impl From<CompositeGlyph> for Glyph {
-    fn from(value: CompositeGlyph) -> Self {
-        Glyph::Composite(value)
+    pub(crate) fn new_composite(glyph_name: GlyphName, composite: CompositeGlyph) -> Glyph {
+        Glyph::Composite(glyph_name, composite)
+    }
+
+    pub(crate) fn glyph_name(&self) -> &GlyphName {
+        match self {
+            Glyph::Simple(name, _) | Glyph::Composite(name, _) => &name,
+        }
     }
-}
 
-impl Glyph {
     pub fn to_bytes(&self) -> Vec<u8> {
         match self {
-            Glyph::Simple(table) => dump_table(table),
-            Glyph::Composite(table) => dump_table(table),
+            Glyph::Simple(_, table) => dump_table(table),
+            Glyph::Composite(_, table) => dump_table(table),
         }
         .unwrap()
     }
 
     pub fn bbox(&self) -> Bbox {
         match self {
-            Glyph::Simple(table) => table.bbox,
-            Glyph::Composite(table) => table.bbox,
+            Glyph::Simple(_, table) => table.bbox,
+            Glyph::Composite(_, table) => table.bbox,
         }
     }
 
     pub fn is_empty(&self) -> bool {
         match self {
-            Glyph::Simple(table) => table.contours().is_empty(),
-            Glyph::Composite(table) => table.components().is_empty(),
+            Glyph::Simple(_, table) => table.contours().is_empty(),
+            Glyph::Composite(_, table) => table.components().is_empty(),
         }
     }
 }
 
+impl IdAware<AnyWorkId> for Glyph {
+    fn id(&self) -> AnyWorkId {
+        AnyWorkId::Be(WorkId::GlyfFragment(self.glyph_name().clone()))
+    }
+}
+
+impl Persistable for Glyph {
+    fn read(from: &mut dyn Read) -> Self {
+        todo!()
```

Comment:
was this intended to be filled in by now?

Reply:
It is on my local copy :D